### ISSUE TITLE
IPropertyAccessOperation incorrectly generated for invalid code

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#31481]
```VB.NET
        Option Explicit On
        Imports System
        Module M1
            Class MyAttr
                Inherits Attribute
                Public Property Prop(ByVal i As Integer) As Integer
                    Get
                        Return Nothing
                    End Get
                    Set(ByVal Value As Integer)
                    End Set
                End Property
            End Class
            &lt;MyAttr(Prop:=1)&gt;'BIND:"Prop"
            Class C1
            End Class
        End Module
```

This results in an IOperation tree with an `IPropertyReferenceOperation` in it. This is because we actually generate a `BoundPropertyAccess` during attribute binding. We should detect this case and generate an `IInvalidOperation` here.
